app/com/arpnetworking/metrics/portal/scheduling/JobExecutorActor.java

```diff
@@ -0,0 +1,180 @@
+
+    /**
+     * Props factory.
+     *
+     * @param <T> The type of result produced by the {@link JobRef}'s job.
+     * @param injector The Guice injector to use to load the {@link JobRepository} referenced by the {@link JobRef}.
+     * @param jobRef The job to intermittently execute.
+     * @return A new props to create this actor.
+     */
+    public static <T> Props props(final Injector injector, final JobRef<T> jobRef) {
+        return props(injector, jobRef, Clock.systemUTC());
+    }
+
+    /**
+     * Props factory.
+     *
+     * @param <T> The type of result produced by the {@link JobRef}'s job.
+     * @param injector The Guice injector to use to load the {@link JobRepository} referenced by the {@link JobRef}.
+     * @param jobRef The job to intermittently execute.
+     * @param clock The clock the scheduler will use, when it ticks, to determine whether it's time to run the next job(s) yet.
+     * @return A new props to create this actor.
+     */
+    protected static <T> Props props(final Injector injector, final JobRef<T> jobRef, final Clock clock) {
+        return Props.create(JobExecutorActor.class, () -> new JobExecutorActor<>(injector, jobRef, clock));
+    }
+
+    private JobExecutorActor(final Injector injector, final JobRef<T> jobRef, final Clock clock) {
+        _injector = injector;
+        _jobRef = jobRef;
+        _clock = clock;
+        _periodicMetrics = new TsdPeriodicMetrics.Builder()
+                .setMetricsFactory(injector.getInstance(MetricsFactory.class))
+                .build();
+    }
+
+    @Override
+    public void preStart() throws Exception {
+        super.preStart();
+        timers().startSingleTimer("TICK", Tick.INSTANCE, TICK_INTERVAL);
+    }
+
+    /**
+     * Returns the time until the actor should next wake up.
+     *
+     * @param now The current time.
+     * @param timeToAwaken The next time the job should run.
+     * @return The time until we should wake up. Non-negative.
+     */
+    /* package private */ static FiniteDuration timeUntilNextTick(final Instant now, final Instant timeToAwaken) {
+        final FiniteDuration delta = Duration.fromNanos(ChronoUnit.NANOS.between(now, timeToAwaken));
+        if (delta.lt(Duration.Zero())) {
+            return Duration.Zero();
+        } else if (delta.lt(TICK_INTERVAL)) {
+            return delta;
+        } else {
+            return TICK_INTERVAL;
+        }
+    }
+
+    private void execute(
+            final JobRepository<T> repo,
+            final Organization org,
+            final Job<T> job,
+            final Instant scheduled
+    ) {
+        job.execute(getSelf(), scheduled)
+                .handle((result, error) -> {
+                    if (error == null) {
+                        repo.jobSucceeded(job.getId(), org, scheduled, result);
+                    } else {
+                        repo.jobFailed(job.getId(), org, scheduled, error);
+                    }
+                    return null;
```

Comment:
When the job is finished how is the next scheduled instance run? -- see comment below about the race condition and avoiding multiple executions for the same scheduled target.